core/src/bundle/usBundleContext.cpp

```diff
@@ -169,9 +176,12 @@ struct ServiceHolder
     {
       bc->UngetService(sref);
     }
-    catch (const std::exception& exp)
+    catch (const std::exception& )
     {
-      US_INFO << "UngetService threw an exception - " << exp.what();
+	  // don't throw exceptions from the destructor. For an explanation, see:
+	  // https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md
+	  // Following this rule means that a FrameworkEvent isn't an option here 
+	  // since it contains an exception object which clients could throw.
```

Comment:
You're right, this wasn't throwing before.

As long as the logger remains accessible through the ```BundleContext``` instance, I can log the exception similar to before. I will do that once the final lifecycle changes are merged into this branch.